test/integration/client-side-encryption/driver.test.ts

```diff
@@ -401,6 +413,176 @@ describe('Client Side Encryption Functional', function () {
       });
     }
   );
+
+  describe('CSOT on ClientEncryption', function () {
+    const metadata: MongoDBMetadataUI = {
+      requires: { clientSideEncryption: true, mongodb: '>=4.4' }
+    };
+    function makeBlockingFailFor(command: string, blockTimeMS: number) {
+      beforeEach(async function () {
+        const utilClient = this.configuration.newClient();
+        await utilClient.db('admin').command({
+          configureFailPoint: 'failCommand',
+          mode: { times: 2 },
+          data: {
+            failCommands: [command],
+            blockConnection: true,
+            blockTimeMS,
+            appName: 'clientEncryption'
+          }
+        } as FailPoint);
+        await utilClient.close();
+      });
+
+      afterEach(async function () {
+        sinon.restore();
+        const utilClient = this.configuration.newClient();
+        await utilClient
+          .db('admin')
+          .command({ configureFailPoint: 'failCommand', mode: 'off' } as FailPoint);
+        await utilClient.close();
+      });
+    }
+
+    function expectCSOTTimeout(fn: () => Promise<void>) {
+      return async () => {
+        const start = performance.now();
+        const error = await fn().then(
+          () => null,
+          error => error
+        );
+        const end = performance.now();
+        if (error?.name === 'MongoBulkWriteError') {
+          expect(error)
+            .to.have.property('errorResponse')
+            .that.is.instanceOf(MongoOperationTimeoutError);
+        } else {
+          expect(error).to.be.instanceOf(MongoOperationTimeoutError);
+        }
+        expect(end - start).to.be.within(498, 1000);
+      };
+    }
+
+    let client: MongoClient;
+    let clientEncryption: ClientEncryption;
+
+    beforeEach(async function () {
+      if (!this.configuration.clientSideEncryption.enabled) {
+        this.skip();
+      }
+
+      client = this.configuration.newClient({}, { appName: 'clientEncryption' });
+      await client.connect();
+      clientEncryption = new ClientEncryption(client, {
+        kmsProviders: { local: { key: LOCAL_KEY } },
+        keyVaultNamespace,
+        keyVaultClient: null,
+        timeoutMS: 500,
+        ...getEncryptExtraOptions()
+      });
+    });
+
+    afterEach(async function () {
+      await client?.close();
+    });
+
+    describe('rewrapManyDataKey', function () {
+      makeBlockingFailFor('update', 2000);
```

Comment:
we’re missing a tests that cover: 
- timeoutMS is respected by the state machine
- timeoutMS applies to the state machine.execute() and bulkWrite collectively

We can probably test this by putting a failCommand on both a find and bulkWrite, such that the total blockTimeMS > timeoutMS.  Neither would fail individually, but it fails if timeoutMS is shared across both.